pkg/progress/progress_watcher.go

```diff
@@ -0,0 +1,124 @@
+package progress
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Watcher struct {
+	progressPercents int
+	exitCode         int
+	buf              *bufio.Reader
+	subscribers      []func(ProgressState)
+	done             chan bool
+	cmd              *exec.Cmd
+}
+
+func CmdWithProgressWatching(name string, arg ...string) (*Watcher, error) {
+	cmd := exec.Command(name, arg...)
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Watcher{
+		progressPercents: 0,
+		buf:              bufio.NewReader(stdout),
+		subscribers:      []func(ProgressState){},
+		done:             make(chan bool),
+		cmd:              cmd,
+	}, nil
+}
+
+func (p *Watcher) Watch() {
+	// track also if exec is not finished to stop automatically
+	go func() {
+		err := p.cmd.Wait()
+
+		exitCode := p.cmd.ProcessState.ExitCode()
+		p.stop(exitCode)
+
+		if err != nil {
+			//TODO Make sure that it's right way to finish the process
+			log.Panic(err)
+		}
+	}()
+
+	ticker := time.NewTicker(500 * time.Millisecond)
+
+	for {
+		select {
+		case <-ticker.C:
+			{
+				line, err := p.buf.ReadSlice('\r')
+				if err != nil {
+					fmt.Printf("Failed to read the next line. Cause: %s", err)
+					//TODO continue or break here? Should we send ERROR
```

Comment:
Please address that TODO

Reply:
Fixed, here I think we can continue restore process